sql-plugin/src/main/spark320/scala/com/nvidia/spark/rapids/shims/HybridFileSourceScanExecMeta.scala

```diff
@@ -0,0 +1,216 @@
+        // For the time being, the native backend may return incorrect results over nestedMap
+        case MapType(kt, vt, _) if kt.isInstanceOf[MapType] || vt.isInstanceOf[MapType] => false
+        // For the time being, BinaryType is not supported yet
+        case _: BinaryType => false
+        case _ => true
+      })
+    }
+    // TODO: supports BucketedScan
+    lazy val noBucketedScan = !fsse.bucketedScan
+    // TODO: supports working along with Alluxio
+    lazy val noAlluxio = !AlluxioCfgUtils.enabledAlluxioReplacementAlgoConvertTime(conf)
+
+    isEnabled && isParquet && allSupportedTypes && noBucketedScan && noAlluxio
+  }
+
+  /**
+   * Check if runtimes are satisfied, including:
+   * - Spark distribution is not CDH or Databricks
+   * - Hybrid jar in the classpath
+   * - Java version is 1.8
+   * - Scala version is 2.12
+   * - Parquet V1 data source
+   */
+  def checkRuntimes(v1DataSourceList: String): Unit = {
+    checkNotRunningCDHorDatabricks()
+    checkHybridJarInClassPath()
+    checkJavaVersion()
+    checkScalaVersion()
+    checkV1Datasource(v1DataSourceList)
+  }
+
+  /**
+   * Check Spark distribution is not CDH or Databricks,
+   * report error if it is
+   */
+  private def checkNotRunningCDHorDatabricks(): Unit = {
+    if (VersionUtils.isCloudera || VersionUtils.isDataBricks) {
+      throw new RuntimeException("Hybrid feature does not support Cloudera/Databricks " +
+          "Spark releases, Please disable Hybrid feature by setting " +
+          "spark.rapids.sql.parquet.useHybridReader=false")
+    }
+  }
+
+  /**
+   * Check if the Hybrid jar is in the classpath,
+   * report error if not
+   */
+  private def checkHybridJarInClassPath(): Unit = {
+    try {
+      Class.forName(HYBRID_JAR_PLUGIN_CLASS_NAME)
+    } catch {
+      case e: ClassNotFoundException => throw new RuntimeException(
+        "Hybrid jar is not in the classpath, Please add Hybrid jar into the class path, or " +
+            "Please disable Hybrid feature by setting " +
+            "spark.rapids.sql.parquet.useHybridReader=false", e)
+    }
+  }
+
+  /**
+   * Hybrid feature only supports 1.8 Java version,
+   * report error if not
+   */
+  private def checkJavaVersion(): Unit = {
+    val javaVersion = System.getProperty("java.version")
+    if (javaVersion == null) {
+      throw new RuntimeException("Hybrid feature: Can not read java.version, get null")
+    }
+    if (!javaVersion.startsWith("1.8")) {
+      throw new RuntimeException(
+        "Hybrid jar is not in the classpath, Please add Hybrid jar into the class path, or " +
+            "Please disable Hybrid feature by setting " +
+            "spark.rapids.sql.parquet.useHybridReader=false")
+    }
+  }
+
+  /**
+   * Hybrid feature only supports Scala 2.12 version,
+   * report error if not
+   */
+  private def checkScalaVersion(): Unit = {
+    val scalaVersion = scala.util.Properties.versionString
+    if (!scalaVersion.startsWith("version 2.12")) {
```

Comment:
We already have shims and a separate jar for scala 2.13.  @gerashegalov is there a way for us to have scala 2.13 specific code that would just fail instead of doing a check like this?